Create README.md (#25)
ACCESS-NRI/umfile_utils: README.md

# UMfile_utils

> UMfile_util is a Python library that provides specific tools to edit UM dump and field files.

Reviewer: ```suggestion
UMfile_util is a Python library that provides specific tools to process [UM files](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf).
```

> # UMfile_utils Desciption

Reviewer: ```suggestion
## List of functions
```

> This package will include three abilities

Reviewer: ```suggestion
- [perturbIC](#perturbic)
- [um_fields_subset](#um-fields-subset)
- [change_date](#change-date)
```

> ### perturbIC.py 

Reviewer: ```suggestion
### perturbIC
```
With the packaging, there will be a Command Line entry point called `perturbIC` (without needing to run it as a python script directly)

> If the simulation fails, this code allows for minor random perturbation of the input theta (surface termperature) in the restart files. 

Reviewer: ```suggestion
The `perturbIC` function applies a random perturbation to a restart file, with an optional seed to control the random generation. 
This can be useful for simulations that fail due to random divergence or for generating multiple ensemble members in climate experiments.

Run `pertubIC --help` for detailed usage information.
```
More information with examples should be provided within the [`description`](https://github.com/ACCESS-NRI/umfile_utils/blob/ceb50fe0a63989ed70077eeb208ae6f805c4522e/src/umfile_utils/perturbIC.py#L25) of the `perturbIC.py` parser.

> Then the simulation can be rerun to avoid crashing. In the future functionality will be added to perturb other fields as well. 

Reviewer: ```suggestion
```

> ### um_fields_subset.py

Reviewer: ```suggestion
### um_fields_subset
```

> This code runs on a Dump file and allows the user to select a group of fields in the file to store 
> in a new fields file or exclude a group of fields from the file. Both of these functionalities will 
> create a new output file. 

Reviewer: ```suggestion
Subset a UM file to generate an output containing only selected fields. Options are available to include or exclude specific STASH variables.
```

> NEED TO ADD USE CASES

Reviewer: ```suggestion

Run `um_fields_subset --help` for detailed usage information.
```
More information to be added within the parser, same as perturbIC

> ### change_date.py
> NEED TO ADD DESCRIPTION 
> 
> 
> Currently these files only operate on Dump files and include a work around to deal with the river field grids.

Reviewer: ```suggestion
### change_date
Change the date of a UM restart file. 

Run `change_date --help` for detailed usage information.
```
More information to be added within the parser, same as perturbIC

## Installation

> DICUSS AT SOME POINT

Reviewer: ```suggestion
TO BE ADDED
```

> ## Usage
> 
> ```python
> If the user wants to run perturbIC with a specific output filename
> python perturbIC input_file_path -a 0.01 -s 2234 -o output_file_path
> 
> Otherwise it is optional perturbIC will create a filename from the input path
> python perturbIC input_file_path -a 0.01 -s 2234
> 
> The amplitude and seed are also optional as well
> python perturbIC input_file_path -a 0.01 -s 2234
> 
> To take a subset of field the user must provide a either a list of fields to exculde (-x)
> python um_fields_subset.py input_file_path -x 155,156,3100,3101 
> 
> To take a subset of field the user must provide a either a list of fields to include (-v)
> python um_fields_subset.py input_file_path -v 155,156,3100,3101
> 
> Or you can choose to have the program to only include pronostics -p
> python um_fields_subset.py input_file_path -p
> 
> These three options must be run seperately

Reviewer: ```suggestion
```
Examples like these should instead be included directly within the related parser, so they show up when `<function> --help` is run`.

```

## Contributing

> 

Reviewer: ```suggestion
TO BE ADDED
```

## License

> 
> NEED TO ADD THE LICENSE

Reviewer: ```suggestion
TO BE ADDED
```
